### PR TITLE
lightning: fix duplicated engine counter metrics

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve? 
Fix duplicated engine counter like following:
```
[2021/03/02 18:10:06.078 +08:00] [INFO] [restore.go:961] [progress] [total=100.0%] [tables="1/1 (100.0%)"] [chunks="100/100 (100.0%)"] [engines="6/3 (200.0%)"] [speed(MiB/s)=35.39753062725018] [state=post-processing]
```

### What is changed and how it works?
- Fix lightning engine finished counter
- Move the `RestoreController.postProcessLock` into import because it only affects importEngine for import backend

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release Note

 - No release note

### CODE REVIEW

**Review by overvenus — COMMENTED** [review]
How to test?

**Comment by glorv** [comment]
> How to test?

Have no good idea, @kennytm  what do you think?

File: pkg/lightning/restore/restore.go [hunk 1]
```diff
@@ -1353,15 +1352,7 @@ func (t *TableRestore) restoreEngines(ctx context.Context, rc *RestoreController
 	if cp.Status < CheckpointStatusIndexImported {
 		var err error
 		if indexEngineCp.Status < CheckpointStatusImported {
-			// the lock ensures the import() step will not be concurrent.
-			if !rc.isLocalBackend() {
-				rc.postProcessLock.Lock()
-			}
 			err = t.importKV(ctx, closedIndexEngine, rc, indexEngineID)
-			rc.saveStatusCheckpoint(t.tableName, indexEngineID, err, CheckpointStatusImported)
```
[the remark this answers is not in the document]
Comment (glorv) [reply pkg/lightning/restore/restore.go:1410]: yes..😂

[the remark this answers is not in the document]
Comment (glorv) [reply pkg/lightning/restore/restore.go:1410]: There is already a call in `importKV`..

**Comment by kennytm** [comment]
> How to test?

Maybe it is possible to use a unit test from `restore_test.go`, and use `metric.ReadCounter` to verify the number at the end.

**Pull request merged**